src/plugins/intel_gpu/src/graph/graph_optimizer/mark_shape_of_subgraphs.cpp

```diff
@@ -107,7 +108,7 @@ void mark_shape_of_subgraphs::mark_node(program_node& node) {
     // Update impl if needed
     const auto default_subgraph_impl = impl_types::cpu;
     if (_update_impls)
-        if (!node.is_type<reshape>())
+        if (!node.is_type<reshape>() && !node.is_type<strided_slice>())
```

Comment:
I think the issue not with StridedSlice itself, but rather with its large input buffer. Excluding StridedSlice from ShapeOf-subgraphs could potentially cause performance issues for other model
We could probably choose some acceptable buffer size for StridedSlice instead, something like in `mark_shape_of_subgraphs::can_mark_node()`:
```
    if (node.is_type<strided_slice>() &&
        node.get_dependency(0).is_constant() && node.get_dependency(0).get_output_layout().count() > 256) {
        return false;
    }
```